pysal/esda/mapclassify.py

```diff
@@ -1093,16 +1095,29 @@ def _set_bins(self):
 
         x = self.y.copy()
         k = self.k
-        res0 = natural_breaks(x, k)
-        fit = res0[2].sum()
-        for i in xrange(self.initial):
-            res = natural_breaks(x, k)
-            fit_i = res[2].sum()
-            if fit_i < fit:
-                res0 = res
-        self.bins = np.array(res0[-1])
-        self.k = len(self.bins)
-        self.iterations = res0[-2]
+        values = np.array(x)
```

Comment:
Would it be worth it in the future to break this off into a well-defined method, like `_can_classify(self, n)`? 

I imagine consistency checks like this logic may need to exist in `_set_bins` for each classifier (e.g. in quantiles if `k > np.unique(data)`).